opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/export/__init__.py

```diff
@@ -268,6 +268,11 @@ def _receive_metrics(
     ) -> None:
         """Called by `MetricReader.collect` when it receives a batch of metrics"""
 
+    @final
+    def force_flush(self, timeout_millis: float = 10_000) -> bool:
+        self.collect(timeout_millis=timeout_millis)
+        return True
```

Comment:
Should the collect be modified to return the bool if the the collection is completed within timeout? Or it's not possible to cancel and return if the timeout exceeded.

Reply:
Hm, not so sure that we should modify collect. It is an interface that is not expected to return anything, the return of a boolean value is something pertinent to `force_flush` on the other hand.

Reply:
>Collect SHOULD provide a way to let the caller know whether it succeeded, failed or timed out. When the Collect operation fails or times out on some of the instruments, the SDK MAY return successfully collected results and a failed reasons list to the caller.

I think we may have missed this initially https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#collect

Reply:
hm, good point, investigating...

Reply:
I submitted a possible solution, PTAL @srikanthccv

Reply:
Please add a docstring that says `collect` raise exception for timeout/error

Reply:
Fixed